src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,101 @@
-import { useState } from 'react';
+import { FC, useState } from 'react';
+import { Movie } from '../../types/Movie';
 import { TextField } from '../TextField';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const data = {
+    title: title.trim(),
+    description: description.trim(),
+    imgUrl: imgUrl.trim(),
+    imdbUrl: imdbUrl.trim(),
+    imdbId: imdbId.trim(),
+  };
+
+  const checkEnteredValue = () => {
+    return Boolean(data.title && data.imdbId && data.imdbUrl && data.imgUrl);
+  };
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+
+    if (!checkEnteredValue()) {
+      return;
+    }
+
+    onAdd(data);
+
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImdbUrl('');
+    setImdbId('');
+
+    setCount(v => v + 1);
```

Comment:
```suggestion
    setCount(prev => prev + 1);
```
What is `v`? Avoid one-letter names